src/main/java/net/openhft/chronicle/core/Jvm.java

```diff
@@ -811,7 +812,7 @@ public static Map<ExceptionKey, Integer> recordExceptions(boolean debug, boolean
     public static Map<ExceptionKey, Integer> recordExceptions(final boolean debug,
                                                               final boolean exceptionsOnly,
                                                               final boolean logToSlf4j) {
-        final Map<ExceptionKey, Integer> map = Collections.synchronizedMap(new LinkedHashMap<>());
+        final Map<ExceptionKey, Integer> map = new ConcurrentSkipListMap<>(Comparator.comparing(ExceptionKey::nanoTimestamp));
```

Comment:
If we use a wrapped LinkedHashMap, the iterator returned from entrySet() won't be tolerate concurrent modification, but if we use a truly concurrent map, the iterator will. Unfortunately there's no ordered concurrent map, only sorted concurrent maps, so if we add a nano timestamp we'll get the ordering by insertion time we're after?

Reply:
I wonder though if inserting sorted data (the timestamps will be sequential naturally) ends up producing some kind of pathological skip list.